### PR TITLE
docs: update ``usage/static_files``

### PR DESCRIPTION
**Pull request by JacobCoffee**
## Description

- Updates ``usage/static_files`` as part of #3212 work

### CODE REVIEW

File: docs/usage/static-files.rst [hunk 1]
```diff
@@ -91,26 +91,23 @@ with support for popular cloud providers available via 3rd party implementations
...
 
-
 .. literalinclude:: /examples/static_files/upgrade_from_static_1.py
-    :language: python
-
+    :caption: Using the deprecated :class:`StaticFilesConfig <litestar.static_files.config.StaticFilesConfig>`
 
 .. literalinclude:: /examples/static_files/upgrade_from_static_2.py
-    :language: python
+    :caption: Upgrading from :class:`StaticFilesConfig <litestar.static_files.config.StaticFilesConfig>` to
```
Comment (JacobCoffee) [inline docs/usage/static-files.rst:112]:
```suggestion
    :caption: Upgrading from :class:`~.static_files.config.StaticFilesConfig` to
```

File: docs/usage/static-files.rst [hunk 2]
```diff
@@ -91,26 +91,23 @@ with support for popular cloud providers available via 3rd party implementations
...
-
 Existing code can be upgraded to :func:`create_static_files_router` by replacing
 :class:`StaticFilesConfig` instances with this function call and passing the result to
 ``route_handlers`` instead of ``static_files_config``:
 
-
 .. literalinclude:: /examples/static_files/upgrade_from_static_1.py
-    :language: python
-
+    :caption: Using the deprecated :class:`StaticFilesConfig <litestar.static_files.config.StaticFilesConfig>`
```
Comment (JacobCoffee) [inline docs/usage/static-files.rst:109]:
```suggestion
    :caption: Using the deprecated :class:`~.static_files.config.StaticFilesConfig`
```

File: docs/usage/static-files.rst [hunk 3]
```diff
@@ -68,14 +71,11 @@ Retrieving paths to static files
 under which a specific file will be available:
 
 .. literalinclude:: /examples/static_files/route_reverse.py
-    :language: python
-
-.. tip::
+    :caption: Retrieving paths to static files using :meth:`Litestar.route_reverse <litestar.app.Litestar.route_reverse>`
```
Comment (JacobCoffee) [inline docs/usage/static-files.rst:74]:
```suggestion
    :caption: Retrieving paths to static files using :meth:`~.app.Litestar.route_reverse`
```

File: docs/usage/static-files.rst [hunk 4]
```diff
@@ -47,18 +50,18 @@ Options available on :class:`~litestar.router.Router` can be passed to directly
...
-
+    :caption: Passing options to the router generated by
+      :func:`create_static_files_router`
 
 Using a custom router class
 ---------------------------
 
-The router class used can be customized with the ``router_class`` parameter:
+The router class used can be customized with the
+:paramref:`~litestar.static_files.create_static_files_router.params.router_class` parameter:
```
Comment (JacobCoffee) [inline docs/usage/static-files.rst:60]:
```suggestion
:paramref:`~.static_files.create_static_files_router.params.router_class` parameter:
```

File: docs/usage/static-files.rst [hunk 5]
```diff
@@ -1,44 +1,47 @@
 Static files
 ============
 
-To serve static files (i.e. serve arbitrary files from a given directory), the
+To serve static files (e.g., serve arbitrary files from a given directory), the
```
Comment (provinzkraut) [inline docs/usage/static-files.rst:4]: I think `i.e.` is correct here. Serving arbitrary files isn't an example of using static files; It's the concept, just a bit more wordy.

**Review by provinzkraut — APPROVED** [review]

File: docs/usage/static-files.rst (hunk 5 above)
Comment (JacobCoffee) [inline docs/usage/static-files.rst:4]:
```suggestion
To serve static files (i.e., serve arbitrary files from a given directory), the
```

**Pull request merged**